File: backend/src/main/java/com/smartrecord/service/RoomService.java
package com.smartrecord.service;

import com.smartrecord.dto.room.CreateRoomReq;
import com.smartrecord.dto.room.JoinRoomReq;
import com.smartrecord.dto.room.RoomResp;
import com.smartrecord.dto.room.UpdateSettingsReq;

import java.util.List;

public interface RoomService {

    RoomResp createRoom(Long userId, CreateRoomReq req);

    RoomResp joinRoom(Long userId, JoinRoomReq req);

    RoomResp getRoomDetail(Long roomId);

    List<RoomResp> getMyRooms(Long userId);

    void quitRoom(Long userId, Long roomId);

    void dissolveRoom(Long userId, Long roomId);

    List<RoomResp> getHistory(Long userId);

    void updateSettings(Long userId, Long roomId, UpdateSettingsReq req);
}

File: miniprogram/services/room-service.js
/**
 * 编队（房间）服务 — 封装房间相关 API
 */
const { get, post, put, del } = require('../utils/request');

/** 获取我的房间列表 */
function getMyRooms() {
  return get('/room/my');
}

/** 获取房间详情 */
function getRoomDetail(roomId) {
  return get(`/room/${roomId}`);
}

/** 创建房间 */
function createRoom(payload) {
  return post('/room', payload).then(res => {
    try {
      const behaviorLogger = require('../utils/behavior-logger');
      behaviorLogger.track('FLEET_CREATE', {
        scoreMode: payload.scoreMode,
        roundInputMethod: payload.roundInputMethod,
        roomId: res ? res.id : ''
      });
    } catch(e) {}
    return res;
  });
}

/** 加入房间 */
function joinRoom(roomNo) {
  return post('/room/join', { roomNo }).then(res => {
    try {
      const behaviorLogger = require('../utils/behavior-logger');
      behaviorLogger.track('FLEET_JOIN', {
        roomNo: roomNo,
        roomId: res ? res.id : ''
      });
    } catch(e) {}
    return res;
  });
}

/** 退出房间 */
function quitRoom(roomId) {
  return del(`/room/${roomId}/quit`);
}

/** 历史房间 */
function getHistory() {
  return get('/room/history');
}

/** 更新记分设置 */
function updateSettings(roomId, settings) {
  return put(`/room/${roomId}/settings`, settings);
}

module.exports = {
  getMyRooms,
  getRoomDetail,
  createRoom,
  joinRoom,
  quitRoom,
  getHistory,
  updateSettings,
};
